crates/polars-plan/src/plans/optimizer/predicate_pushdown/mod.rs

```diff
@@ -632,34 +657,32 @@ impl<'a> PredicatePushDown<'a> {
                 acc_predicates,
             ),
             lp @ Union { .. } => {
-                let mut local_predicates = vec![];
-
-                // a count is influenced by a Union/Vstack
-                acc_predicates.retain(|_, predicate| {
-                    if has_aexpr(predicate.node(), expr_arena, |ae| matches!(ae, AExpr::Len)) {
-                        local_predicates.push(predicate.clone());
-                        false
-                    } else {
-                        true
+                if cfg!(debug_assertions) {
+                    for v in acc_predicates.values() {
+                        let ae = expr_arena.get(v.node());
+                        assert!(permits_filter_pushdown(
```

Comment:
drive-by - predicates that hit here should already satisfy these requirements - otherwise they should have been caught at the `Filter` node